include/seastar/util/memory_diagnostics.hh

```diff
@@ -74,12 +78,22 @@ using memory_diagnostics_writer = noncopyable_function<void(std::string_view)>;
 ///     the output.
 void set_additional_diagnostics_producer(noncopyable_function<void(memory_diagnostics_writer)> producer);
 
-/// Manually generate a diagnostics report
+/// Generate and return a diagnostics report as a string.
 ///
 /// Note that contrary to the automated report generation (triggered by
 /// allocation failure), this method does allocate memory and can fail in
 /// low-memory conditions.
 sstring generate_memory_diagnostics_report();
 
+namespace internal {
```

Comment:
I put it in internal "just because" I'm only using it in tests and it allows us to change this in the future, but feedback is especially welcome on this decision.

Reply:
We try to increase our flexibility to making API changes, so placing things in the internal namespace is very welcome.